### PR TITLE
[ISSUE-84] Fixup bug which made CPT could not predict a value already in target sequence

### PR DESCRIPTION
**Pull request by bluesheeptoken**
Bug fix where subsequent of `BAAC` should be `AC` and not `C`.
Full story [here](https://medium.com/@rafaelktistakis/just-reading-through-the-comments-and-the-question-raised-by-christopher-a78295440baf)

### CODE REVIEW

File: tests/test_cpt.py [hunk 1]
```diff
@@ -38,15 +38,15 @@ def test_predict(self):
...
+        self.assertEqual(self.cpt.predict([['A'], ['A', 'B']]), ['B', 'C'])
         # Test if predictions are the same with multi threading turned off
-        self.assertEqual(self.cpt.predict([['A'], ['A', 'B']], False), ['B', 'D'])
+        self.assertEqual(self.cpt.predict([['A'], ['A', 'B']], False), ['B', 'C'])
 
         self.cpt.MBR = 2
         self.assertEqual(self.cpt.predict([['A', 'B']]), ['C'])
 
         self.cpt.noise_ratio = 0.2
-        self.cpt.MBR = 1
```
Comment (bluesheeptoken) [inline tests/test_cpt.py:49]: Regarding the definition of the `MBR`, we want to have 2 sequence to match here.

File: tests/test_cpt.py [hunk 2]
```diff
@@ -38,15 +38,15 @@ def test_predict(self):
 
         self.cpt.noise_ratio = 1
         self.cpt.MBR = 3
-        self.assertEqual(self.cpt.predict([['A'], ['A', 'B']]), ['B', 'D'])
+        self.assertEqual(self.cpt.predict([['A'], ['A', 'B']]), ['B', 'C'])
         # Test if predictions are the same with multi threading turned off
-        self.assertEqual(self.cpt.predict([['A'], ['A', 'B']], False), ['B', 'D'])
```
[the remark this answers is not in the document]
Comment (catchthemonster) [reply tests/test_cpt.py:43]: Hi, which branch this fix in ...
i would like to clone this branch ...

[the remark this answers is not in the document]
Comment (bluesheeptoken) [reply tests/test_cpt.py:43]: Hey,
This bug was due to how I computed MBR.

I planned to merge this after the other branch which fixes the CI. But I can merge it now if you want to.

PS: Sorry for the noise with the other PR ^^'. I am trying to limit the noise.

[the remark this answers is not in the document]
Comment (catchthemonster) [reply tests/test_cpt.py:43]: no issue, which branches are we talking about here?
i am a bit confused...
CI is what? sorry to be ignorant. I see that you have something going for the macos and github-actions... 
I am not sure what is that but would love to test in my code your changes for MBR in CPT.
No issue on the noise, the fact is your made a product, so it is good that people are using it and comment on it ...

[the remark this answers is not in the document]
Comment (bluesheeptoken) [reply tests/test_cpt.py:43]: CI means Continuous Integration, in a few words, a pipeline to test your code at each push / PR / merge etc...

Regarding the MBR thing, I just fixed how it should be computed and fix this test. This test was matching way more than MBR sequences (which was not intended. But I was probably not really focused for this test :) )

The main goal of this PR is to fix an issue about a misunderstanding I had. Everything is explained [here](https://medium.com/@rafaelktistakis/just-reading-through-the-comments-and-the-question-raised-by-christopher-a78295440baf) (Rafael Ktistakis is one of the co-author of CPT).

If you have spare time / real-world use case, do not hesitate to test this branch. If you prefer to wait for pypi package (to just install it with `pip install cpt`), I will probably release it by the end of the week-end

**Pull request merged**